05-Mining-Protocol.md

```diff
@@ -476,7 +476,7 @@ The `mining_job_token` provides the information for the pool to authorize the cu
 ### 5.3.19 `SetCustomMiningJob.Success` (Server -> Client)
 
 Response from the server when it accepts the custom mining job.
-Client can start to mine on the job immediately (by using the `job_id` provided within this response).
+Client can start submitting shares for this job immediately (by using the `job_id` provided within this response).
```

Comment:
Using the `job_id` implies that you can't send shares earlier than this point. Or do you leave job_id blank in that case?

Reply:
`SetCustomMiningJob.Success` is the Pool commitment to this specific `job_id`

the hasher already had all info they needed long ago, so they're not gonna sit and wait... they're gonna buffer that work optimistically, and only start submitting it once Pool committed to rewarding it

that's the point of this clarification, please let me know if you feel this is still not clear enough, or whether you feel this is flawed in some sense

Reply:
Ah, so the miner needs to hold on to the shares it finds until it has the `job_id`, and then submit them all at once? That should be clarified.

Reply:
For good measure, you could say that the backlog of shares should even be submitted if a new a tip was found in the mean time.